.github/workflows/reusable_test.yml

```diff
@@ -0,0 +1,191 @@
+        description: "Mage build target"
+        default: 'prod'
+      memgraph_version:
+        type: string
+        description: "Memgraph version built into this image (format: X.Y.Z)"
+        default: '2.18.0'
+      memgraph_download_link:
+        type: string
+        description: "Memgraph package download link. Leave empty to use the official download link."
+        required: false
+      memgraph_ref:
+        type: string
+        description: "Memgraph submodule branch to build query modules from"
+        default: 'master'
+      memgraph_ref_update:
+        type: string
+        description: "Update Memgraph submodule to the latest commit"
+        default: false
+
+env:
+  PY_VERSION: "3.9"
+  CORE_COUNT: "8"
+  MAGE_CONTAINER: "mage"
+  MEMGRAPH_PORT: 7687
+  NEO4J_PORT: 7688
+  NEO4J_CONTAINER: "neo4j_test"
+  OS: "debian-11"
+
+jobs:
+  BuildAndTest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository and submodules
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          submodules: recursive
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout specific branch in submodule
+        run: |
+          cd cpp/memgraph
+          git checkout ${{ inputs.memgraph_ref }}
+          if [[ "${{ inputs.memgraph_ref_update }}" == "true" ]]; then
+            git pull
+          fi
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set up and check memgraph download link
+        run: |
+          if [[ -z "${{ inputs.memgraph_download_link }}" ]]; then
+            echo "Using official download link"
+            mg_url="https://download.memgraph.com/memgraph/v${{ inputs.memgraph_version }}/$OS${{ inputs.arch == 'arm64' && '-aarch64' || '' }}/memgraph_${{ inputs.memgraph_version }}-1_${{ inputs.arch }}.deb"
+          else
+            echo "Using custom download link"
+            mg_url=${{ inputs.memgraph_download_link }}
+          fi
+          if curl --output /dev/null --silent --head --fail $mg_url; then
+            echo "Memgraph download link is valid"
+            echo "MEMGRAPH_DOWNLOAD_LINK=${mg_url}" >> $GITHUB_ENV 
+          else
+            echo "Memgraph download link is not valid"
+            exit 1
+          fi
+
+      - name: Download memgraph binary
+        run: curl -L ${{ env.MEMGRAPH_DOWNLOAD_LINK }} > memgraph-${{ inputs.arch }}.deb
+
+      - name: Disk status before cleaning
+        run: |
+          df -h
+          docker buildx du
+
+      - name: Docker system prune
+        run: |
+          docker buildx prune --all -f
+          sudo rm -rf /opt/hostedtoolcache/CodeQL
+          sudo rm -rf /opt/hostedtoolcache/go
+          sudo rm -rf /opt/hostedtoolcache
+
+      - name: Disk status after cleaning
+        run: |
+          df -h
+          docker buildx du
```

Comment:
I think you can remove this disk status, it is not used anymore for anything useful

Reply:
Done